core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -175,8 +175,18 @@ public InternalEngine(EngineConfig engineConfig) throws EngineException {
                         throw new IllegalArgumentException(openMode.toString());
                 }
                 logger.trace("recovered [{}]", seqNoStats);
-                indexWriter = writer;
                 seqNoService = sequenceNumberService(shardId, engineConfig.getIndexSettings(), seqNoStats);
+                // norelease
+                /*
+                 * We have no guarantees that all operations above the local checkpoint are in the Lucene commit. These means that we there
```

Comment:
nit: "are in the lucene commit" -> "are in the lucene commit or the translog generation associated with it"